### ISSUE TITLE
Store evaluation result in a register

### ISSUE THREAD

**Issue by nbardiuk** [liquidz/vim-iced#297]
Some times I reach into stdout buffer to copy the result of last evaluation. I was wondering if it is in the scope of the plugin to store evaluation results in a register. 

From the user perspective it could be set up in different ways: 
* optional config to always store the last result in a specific register (like g:iced#eval#mark_at_last)
* a command that reads register letter and evals into it (like eval at mark)
* extend existing command to accept register like yank `"x<Plug>(iced_eval)<motion>` (don't know if it even possible)

**Comment by liquidz** [comment]
@nbardiuk Thanks for your suggestion!
It's very interesting.

> extend existing command to accept register like yank "x<Plug>(iced_eval)<motion> (don't know if it even possible)

I like this one the closest to the Vim way.
I tried to update `<Plug>(iced_eval)` operator experimentally.

Could you try `"x<Leader>ee` (default mapping) or custom mapping like below in [feature/register](https://github.com/liquidz/vim-iced/tree/feature/register) branch?
```vim
" mapping for "xee
nmap <silent> ee <Plug>(iced_eval)<Plug>(sexp_outer_list)``
```

**Comment by nbardiuk** [comment]
Great! I've tested diffferent scenarios

* `nmap ee <Plug>(iced_eval)` and call  `"eeeaf` - stores result in register `e` :+1: 
*  `set clipboard=""` `nmap ee <Plug>(iced_eval)` and call  `eeaf`- does not store result in registers :+1: 
* `set clipboard=unnamedplus` `nmap ee <Plug>(iced_eval)` and call it `eeaf`- stores result in register `+` :question:
* `nmap ee "e<Plug>(iced_eval)` and call `eeaf` - stores result in register `e` :+1: 
*  `nmap ee "e<Plug>(iced_eval)` and call `"xeeaf` - stores result in register `e` :+1: 

Another observation:
 *  successful evaluation shows message with the result :+1: 
 * failed evaluation is silent (no output) :x:

**Comment by liquidz** [comment]
@nbardiuk Thanks for your testing!

> set clipboard=unnamedplus nmap ee <Plug>(iced_eval) and call eeaf- stores result in register + ❓

It seems to be Vim's specification.
https://vim-jp.org/vimdoc-en/eval.html#v:register

> failed evaluation is silent (no output) ❌

Nice catch! I fixed it in `feature/register` branch.

**Comment by liquidz** [comment]
https://github.com/liquidz/vim-iced/blob/feature/register/autoload/iced/operation.vim#L33
For now, only `"` register is skipped.

**Comment by nbardiuk** [comment]
Works great!

I am curious whether you will add register support to other eval operations.
It is not critical, having an eval with register, I can compose eval_and_print for myself .

**Comment by liquidz** [comment]
@nbardiuk Sorry for late reply.
I have no plan to add register support to other operations for now.

**Comment by liquidz** [comment]
I merged to `dev` branch.
I'll cut a new release in this weekend.

**Comment by liquidz** [comment]
Closed by #299
